frappe/core/doctype/log_settings/log_settings.py

```diff
@@ -2,63 +2,143 @@
 # Copyright (c) 2020, Frappe Technologies and contributors
 # License: MIT. See LICENSE
 
+from typing import Protocol, runtime_checkable
+
 import frappe
 from frappe import _
+from frappe.model.base_document import get_controller
 from frappe.model.document import Document
-from frappe.query_builder import DocType, Interval
-from frappe.query_builder.functions import Now
+from frappe.utils import cint
+from frappe.utils.caching import site_cache
+
+DEFAULT_LOGTYPES = {
+	"Error Log": 30,
+	"Activity Log": 90,
+	"Email Queue": 30,
+	"Route History": 90,
+	"Error Snapshot": 30,
+	"Scheduled Job Log": 90,
+}
+
+
+@runtime_checkable
+class LogType(Protocol):
+	"""Interface requirement for doctypes that can be cleared using log settings."""
+
+	@staticmethod
+	def clear_old_logs(days: int) -> None:
```

Comment:
I went with just 1 config parameter for now, IDK if anything else is really "generic" or important. Doctypes can implement those if they wish (like "only clearing successful status")

@gavindsouza  @phot0n

Reply:
Or, only clearing "seen" logs 🤔

Reply:
@gavindsouza we can add filtering for statuses then?  👀 


That's common across most doctypes.